osgar/test_logger.py

```diff
@@ -305,4 +307,149 @@ def test_read_gen_seek(self):
+                    # TODO this write should rise exception as the time gap is too big to track!
+                    t4 = log.write(1, b'\x04')
+                    self.assertEqual(t4, timedelta(hours=4))
+            with LogIndexedReader(log.filename) as log:
+                dt, channel, data = log[1]
+                self.assertEqual(dt, timedelta(hours=0))
+                dt, channel, data = log[2]
+                self.assertEqual(dt, timedelta(hours=1))
+                dt, channel, data = log[3]
+                self.assertEqual(dt, timedelta(hours=2))
+    #            dt, channel, data = next(log)
+    #            self.assertEqual(dt, timedelta(hours=4))
+
+    def _test_large_blocks_with_time_overflow(self):
+        with ExitStack() as at_exit:
+            with patch('osgar.logger.datetime.datetime'):
+                osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1))
+                with osgar.logger.LogWriter(prefix='tmpA', note='test_time_overflow with large blocks') as log:
+                    at_exit.callback(os.remove, log.filename)
+                    t1 = log.write(1, b'\x01'*100000)
+                    self.assertEqual(t1, timedelta(0))
+                    osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1, 1))
+                    t2 = log.write(1, b'\x02'*100000)
+                    self.assertEqual(t2, timedelta(hours=1))
+                    osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1, 2))
+                    t3 = log.write(1, b'\x03'*100000)
+                    self.assertEqual(t3, timedelta(hours=2))
+            with LogIndexedReader(filename) as log:
+                dt, channel, data = log[1]
+                self.assertEqual(dt, timedelta(hours=0))
+                dt, channel, data = log[2]
+                self.assertEqual(dt, timedelta(hours=1))
+                dt, channel, data = log[3]
+                self.assertEqual(dt, timedelta(hours=2))
+
+    def test_no_eof(self):
+        with patch('osgar.logger.datetime.datetime'):
+            osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1))
+            with LogWriter(prefix='tmpEof', note='test_EOF') as log:
+                filename = log.filename
+                osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1, 1))
+                t1 = log.write(1, b'\x01'*100)
+                osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1, 2))
+                t2 = log.write(1, b'\x02'*100)
+                osgar.logger.datetime.datetime = TimeStandsStill(datetime(2019, 1, 1, 3))
+                t3 = log.write(1, b'\x03'*100000)
+
+        partial = filename + '.part'
+        with open(filename, 'rb') as f_in, open(partial, 'wb') as f_out:
+            f_out.write(f_in.read(100))
+            f_out.flush()
+
+            with LogIndexedReader(partial) as log:
+                self.assertEqual( len(log), 1 )
```

Comment:
I used to have spaces around parameters but now I prefer PEP8(?) form "self.assertEqual(len(log), 1)" ... I am not 100% sure ...

Reply:
fixed